### ISSUE TITLE
Generate Test Doesn't Test Temperature Properly

### ISSUE THREAD

**Issue by WilliamParks** [modernaicourse/hw4#2]
`test_generate` and 'submit_generate` don't properly test sampling from the output logits.

ToyModel does the following to generate the test logits, and is used in each test function
```
logits = torch.full((1, tokens.shape[1], self.vocab_size), -1e9)
next_token = self.next_tokens[len(self.calls) - 1]
logits[0, -1, next_token] = 0.0
```

Since all but one of the output logits is -1e9, the tests won't distinguish between using `torch.multinomial` to sample from the output distribution, and just running argmax.
